test/Dialect/Comb/canonicalization.mlir

```diff
@@ -1358,3 +1358,32 @@ hw.module @propagateNamehint(%x: i16) -> (o: i1) {
   %0 = comb.icmp eq %c0_i16, %x {sv.namehint = "hint"}: i16
   hw.output %0 : i1
 }
+
+// CHECK-LABEL: @reductionOrOps
+hw.module @reductionOrOps(%a: i1, %b: i2) -> (c: i1) {
+  // CHECK-NEXT: %c0_i2 = hw.constant 0 : i2
+  // CHECK-NEXT: %0 = comb.icmp ne %b, %c0_i2 {sv.namehint = "_b"} : i2
+  %0 = comb.extract %b from 1 : (i2) -> i1
+  %1 = comb.extract %b from 0 : (i2) -> i1
+  %2 = comb.or %0, %1 {sv.namehint = "_b"} : i1
+  hw.output %2 : i1
+}
+
+// CHECK-LABEL: @reductionAndOps
+hw.module @reductionAndOps(%a: i1, %b: i2) -> (c: i1) {
+  // CHECK-NEXT: %c-1_i2 = hw.constant -1 : i2
+  // CHECK-NEXT: %0 = comb.icmp eq %b, %c-1_i2 {sv.namehint = "_b"} : i2
+  %0 = comb.extract %b from 1 : (i2) -> i1
+  %1 = comb.extract %b from 0 : (i2) -> i1
+  %2 = comb.and %0, %1 {sv.namehint = "_b"} : i1
+  hw.output %2 : i1
+}
+
+// CHECK-LABEL: @reductionXorOps
+hw.module @reductionXorOps(%a: i1, %b: i2) -> (c: i1) {
+  // CHECK-NEXT: %0 = comb.parity %b {sv.namehint = "_b"} : i2
+  %0 = comb.extract %b from 1 : (i2) -> i1
+  %1 = comb.extract %b from 0 : (i2) -> i1
+  %2 = comb.xor %0, %1 {sv.namehint = "_b"} : i1
+  hw.output %2 : i1
+}
```

Comment:
Could you merge tests into one test for simplicity and also remove namehints here? e.g:
```mlir
hw.module @extractToReductionOps(%a: i1, %b: i2) -> (c: i1, d: i1, e: i1) {
  %0 = comb.extract %b from 1 : (i2) -> i1
  %1 = comb.extract %b from 0 : (i2) -> i1
  %2 = comb.or %0, %1 : i1
  %3 = comb.and %0, %1 : i1
  %4 = comb.xor %0, %1 : i1

  hw.output %2, %3, %4 : i1, i1, i1
}
```